virtualpdu/tests/integration/test_entrypoint.py

```diff
@@ -56,25 +61,77 @@ def test_config(self):
         with tempfile.NamedTemporaryFile() as f:
             f.write(bytearray(TEST_CONFIG, encoding='utf-8'))
             f.flush()
-            p = subprocess.Popen([
-                sys.executable,
-                self._get_entrypoint_path('virtualpdu'),
-                f.name
-            ])
+            try:
+                p = subprocess.Popen([
+                    sys.executable,
+                    self._get_entry_point_path('virtualpdu'),
+                    f.name,
+                ])
+
+                self.turn_off_outlet(community='public',
+                                     listen_address='127.0.0.1',
+                                     outlet=5,
+                                     port=9998)
+
+                self.turn_off_outlet(community='public',
+                                     listen_address='127.0.0.1',
+                                     outlet=2,
+                                     port=9997)
+            finally:
+                # NOTE(mmitchell): The process shouldn't have died on it's
+                #                  own. Kill it.
+                p.kill()
+
+    def test_sigint_stops_the_process(self):
+        with tempfile.NamedTemporaryFile() as f:
+            f.write(bytearray(TEST_CONFIG, encoding='utf-8'))
+            f.flush()
+            try:
+                p = subprocess.Popen([
+                    sys.executable,
+                    self._get_entry_point_path('virtualpdu'),
+                    f.name,
+                ])
+
+                self.assertIsNone(p.poll())
+                p.send_signal(signal.SIGINT)
+                time.sleep(0.1)  # NOTE(mmitchell): Give it 100ms to stop.
```

Comment:
you could use retry with a 0.01 sleep and 10 tries, that way if it did indeed stop very fast you don't waste 100 ms